### DISCUSSION TITLE
User Permissions Report asking for client secret

### DISCUSSION THREAD

**Discussion by sarinityx** [Barbarur/NovaPoint#15] [Q&A]
I've filled out the tenant/application ID however it appears it wants a client secret, is there anywhere I can define the parameter?

A configuration issue is preventing authentication - check the error message from the server for details. You can modify the configuration in the application registration portal. See https://aka.ms/msal-net-invalid-client for details.  Original exception: AADSTS7000218: The request body must contain the following parameter: 'client_assertion' or 'client_secret'. Trace ID: 5c244185-1071-4c85-8c46-21944a1a8802 Correlation ID: ba4f5b4d-154e-480e-aafa-f1415f08f761 Timestamp: 2024-05-29 02:39:32Z

**Comment by Barbarur** [comment]
When you register your app in Entra ID and provide API permissions. Did you choose "Delegate" or "Application" permission for the APIs?

This app is designed to manage delegated permissions.
